### PR TITLE
[BEAM-2744] rename BeamRecordType#size()

### PR DESCRIPTION
**Pull request by xumingming**
Summary:

* BeamRecord.size() -> BeamRecord.getFieldCount()
* BeamRecordType.size() -> BeamRecordType.getFieldCount()
* remove BeamRecord.valueInString()

### CODE REVIEW

File: sdks/java/core/src/main/java/org/apache/beam/sdk/values/BeamRecord.java [hunk 1]
```diff
@@ -180,19 +180,7 @@ public boolean isNull(int idx) {
...
-   * Return data fields as key=value.
-   */
-  public String valueInString() {
-    StringBuilder sb = new StringBuilder();
-    for (int idx = 0; idx < size(); ++idx) {
-      sb.append(
-          String.format(",%s=%s", getDataType().getFieldsName().get(idx), getFieldValue(idx)));
-    }
-    return sb.substring(1);
+    return "BeamRecord [dataValues=" + dataValues + ", dataType=" + dataType + "]";
```
Comment (XuMingmin) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/values/BeamRecord.java]: can you also add a `toString()` to `BeamRecordType` and `BeamRecordSqlType`?

Comment (xumingming) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/values/BeamRecord.java]: sure

**Comment by XuMingmin** [comment]
retest this please

**Comment by xumingming** [comment]
Rebased.

**Comment by xumingming** [comment]
R: @takidau @XuMingmin

**Comment by XuMingmin** [comment]
retest this please

**Pull request closed without merging**